Toy2/src/math/Calculation.java

```diff
@@ -0,0 +1,13 @@
+package math;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+public interface Calculation {
+
+    //가변인자-동적상황, List 정적상황 명확한 차이가뭐지?
+    //BigDecimal calculate(BigDecimal... values);
+    BigDecimal calculate(List<BigDecimal> values);
```

Comment:
가변인자로 받을 때의 타입과 List로 명시적으로 파라미터를 정의했을 때 타입은 어떻게 다를까요? 직접 두 타입으로 메서드를 사용해보시면 알 수 있겠네요.

Reply:
연산은 딱 두가지의 숫자만 받아서 수행합니다.
List로 받는다는 뜻은 연산할 숫자 대상이 0개일수도, 1개일수도, 2개 이상일수도 있다는 뜻입니다.
명확하게 두개의 숫자를 받아야하므로 이 경우에는 아래와 같은 시그니처로 만드는게 더 적합합니다.

```suggestion
  BigDecimal calculate(BigDecimal number1, BigDecimal number2);
```

Reply:
가변인자는 배열로 만들어지는데 그러다 보니 배열의 특성을 가지고있고 
최초 가변인자로 2개의 인자를 전달하여 객체를 생성하면 2개의 값을 가지고 있는 배열이 됩니다
그러면 배열의 특성과 같은 최초 호출시에 크기가 정해진다는것이고 배열과 다른점이라고 하면 직접 크기를 지정하지않고
그저 타입에 맞는 인자만 전달해주면 그 갯수에 맞는 배열이 생성되는것 같은데
가변인자가 빛을 발휘하는 상황이 있나요? List에 비해서 메모리의 효율이 좋은거말고는 없는것 같은데
이 상황도 계산기를 예로들면 1 + 1 처럼 2개의 숫자의 연산만 처리하겠다 라는 특수한 조건이 있지않는이상은
보편적으로 사용하지않을것같아서요

Reply:
말씀하신대로 보편적으로 사용하지는 않습니다. 가변인자보다는 명확하게 컬랙션을 넘겨주도록 하는게 좋습니다.
다만 가변인자를 쓰면 배열이 메서드 시작하는 시점에 생기게 되니 메서드 외부에서 해당 배열을 참조할 수 없다는 장점이 있겠으나, 이 또한 컬랙션을 복사하여 넘겨주는 방식을 사용하면 해결 가능합니다~